torchtune/data/_messages.py

```diff
@@ -177,31 +193,42 @@ class InputOutputToMessages(Transform):
 
     def __init__(
         self,
-        train_on_input: bool = False,
+        masking_strategy: Optional[str] = "train_on_assistant",
```

Comment:
Why do we switch the order of these arguments? If someone is passing `InputOutputToMessages(False)` this will break them, right? (Since we unfortunately don't enforce keyword-only args on these APIs). Personally I would leave train_on_input in the first position and put masking_strategy last

Reply:
agreed, let's keep the positions of the arguments the same until we deprecate